docs/integrations/ibis/index.md

```diff
@@ -0,0 +1,268 @@
+) -> ir.Table:
+    """Select feature columns and filter rows"""
+    return feature_table[feature_selection].filter(condition)
+```
+![ibis + hamilton column-level](ibis_column.png)
+
+> 💡 If your code is already structured with Hamilton, migrating from pandas to Ibis should be easy!
+
+### Orchestration
+
+Hamilton is the ideal orchestration partner for Ibis because it's the lightest "orchestrator" you'll find. In fact, it's a simple Python package with 4 dependencies, and you don't need "framework code" to get started, just plain Python functions. When moving to production, Hamilton has all the necessary production-ready features to complement Ibis, such as swapping components, configurations, and lifecycle hooks for logging, alerting, and telemetry.
+
+A simple usage pattern of Hamilton + Ibis is to use the [`@config.when` function modifier](../../concepts/function-modifiers.rst). In the following example, we have alternative implementations for the backend connection, which will be used for computing and storing results.
+
+```python
+# ibis_dataflow.py
+import ibis
+import ibis.expr.types as ir
+from hamilton.function_modifiers import config
+
+# ... entire dataflow definition
+
+@config.when(backend="duckdb")
+def backend_connection__duckdb(
+    connection_string: str
+) -> ibis.backends.BaseBackend:
+    """Connect to DuckDB backend"""
+    return ibis.duckdb.connect(connection_string)
+
+@config.when(backend="bigquery")
+def insert_results__bigquery(
+    project_id: str,
+    dataset_id: str,
+) -> ibis.backends.BaseBackend:
+    """Connect to BigQuery backend
+    Install dependencies via `pip install ibis-framework[bigquery]`
+    """
+    return ibis.bigquery.connect(
+        project_id=project_id,
+        dataset_id=dataset_id,
+    )
+
+def insert_results(
+    backend_connection: ibis.backends.BaseBackend,
+    result_table: ir.Table,
+    table_name: str
+):
+    """Execute expression and insert results"""
+    backend_connection.insert(
+        table_name=table_name,
+        obj=result_table
+    )
+```
+
+```python
+# run.py
+from hamilton import driver
+import ibis_dataflow
+
+dr = (
+    driver.Builder()
+    .with_config({"backend": "duckdb"})
+    .with_modules(ibis_dataflow)
+    .build()
+)
+inputs = dict(connection_string=...)
+# `insert_results` implementation depends on config
+dr.execute(["insert_results"], inputs=inputs)
+```
+
+## How Ibis improves the Hamilton experience
+### Atomic data transformation documentation
+Ibis provides Hamilton users with high-performance local engines (DuckDB, Polars) instead of pandas. Also, Ibis complements Hamilton's dataflow visualization with a more detailed view of the query plan and schemas. See this Ibis visualization for the column-level Hamilton dataflow defined above. It includes all renaming, type casting, and transformations steps. Please zoom in!
+
+![ibis_full_viz](ibis_plan.png)
+
+### Working across rows with user-defined functions (UDFs)
```

Comment:
I'm missing something -- how does this use Hamilton? Maybe have the driver? Or a DAG?

Reply:
added figure. You can apply a function to every row. In a sense, it does the dynamic "Parallelizable/Collect" directly on the backend